### PR TITLE
[zydis] Update version to v4.1.0

### LINKED ISSUE #38691: [zydis] update to 4.1.0
### Library name

zydis

### New version number

4.1.0

### Other information that may be useful (release notes, etc...)

https://github.com/zyantific/zydis/releases/latest

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/issues/38691

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/zydis/zycore.patch [hunk 1]
```diff
@@ -1,66 +1,22 @@
...
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 62d5de6..0f3bcf1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -138,7 +138,7 @@ function (locate_zycore)
+     )
+ endfunction ()
+ 
+-locate_zycore()
++find_package(zycore CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/zydis/zycore.patch:10]:
```suggestion
+find_package(zycore CONFIG REQUIRED)
+add_library(Zycore ALIAS Zycore::Zycore)
```

File: ports/zydis/zycore.patch [hunk 2]
```diff
@@ -1,66 +1,22 @@
...
+ # Library configuration                                                                           #
+@@ -151,7 +151,7 @@ else ()
+     target_compile_definitions("Zydis" PUBLIC "ZYDIS_STATIC_BUILD")
+ endif ()
+ 
+-target_link_libraries("Zydis" PUBLIC "Zycore")
++target_link_libraries("Zydis" PUBLIC Zycore::Zycore)
+ target_include_directories("Zydis"
+     PUBLIC
+         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
```
Comment (dg0yt) [inline ports/zydis/zycore.patch:22]:
```suggestion
```

**Review by dg0yt — COMMENTED** [review]
Untested suggestion.

Bonus question: Does the cmake config do `find_dependency(zycore CONFIG)`?

**Review by dg0yt — COMMENTED** [review]
Portfile bonus question:
What does 
~~~cmake
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
~~~
remove, after
~~~cmake
vcpkg_copy_tools(TOOL_NAMES ZydisDisasm ZydisInfo AUTO_CLEAN)

**Comment by dg0yt** [comment]
https://github.com/FrankXie05/vcpkg/pull/7

**Comment by dg0yt** [comment]
> [FrankXie05#7](https://github.com/FrankXie05/vcpkg/pull/7)

FTR the idea of the patch there is that it might be upstreamed (when they do proper aliasing of targets in the zycore project).

**Comment by dg0yt** [comment]
> FTR the idea of the patch there is that it might be upstreamed (when they do proper aliasing of targets in the zycore project).

https://github.com/zyantific/zycore-c/pull/71  
https://github.com/zyantific/zydis/pull/508

**Comment by dg0yt** [comment]
With my PRs accepted upstream, https://github.com/zyantific/zydis/pull/508/files is what should be applied here.

File: ports/zydis/portfile.cmake [hunk 3]
```diff
@@ -1,27 +1,23 @@
...
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" ZYDIS_BUILD_SHARED_LIB)
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        -DZYAN_SYSTEM_ZYCORE=ON
         -DZYDIS_BUILD_SHARED_LIB=${ZYDIS_BUILD_SHARED_LIB}
+        -DZYDIS_BUILD_DOXYGEN=OFF
         -DZYDIS_BUILD_EXAMPLES=OFF
```
Comment (dg0yt) [inline ports/zydis/portfile.cmake:21]:
```suggestion
        -DZYDIS_BUILD_EXAMPLES=OFF
        -DZYDIS_BUILD_TESTS=OFF
```

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**